commcare/src/main/java/org/motechproject/commcare/tasks/builder/CaseTriggerBuilder.java

```diff
@@ -51,20 +55,30 @@ public CaseTriggerBuilder(CommcareSchemaService schemaService, CommcareConfigSer
     public List<TriggerEventRequest> buildTriggers() {
         List<TriggerEventRequest> triggers = new ArrayList<>();
 
+        Map<String, Set<String>> caseTypes = new HashMap<>();
+
         for (Config config : configService.getConfigs().getConfigs()) {
-            for (Map.Entry<String, Set<String>> entry : schemaService.getAllCaseTypes(config.getName()).entrySet()) {
-                List<EventParameterRequest> parameterRequests = new ArrayList<>();
-                parameterRequests.add(new EventParameterRequest("commcare.field.configName", CONFIG_NAME));
-                addCommonCaseFields(parameterRequests);
+            for (CommcareApplicationJson application : applicationService.getByConfigName(config.getName())) {
+                for (CommcareModuleJson module : application.getModules()) {
+                    if (!caseTypes.containsKey(module.getCaseType())) {
```

Comment:
It just has to check if CaseTrigger with certain CaseType already exists. Changed to Set.